src/components/MemberComponent.js

```diff
@@ -0,0 +1,317 @@
+    (newMember, newState) => ({ ...newMember, ...newState }),
+    initialNewMember
+  );
+
+  useEffect(() => {
+    if (project) {
+      const { memberList } = project;
+      setState({ memberList });
+    } else displayPopup("Project not loaded");
+  }, []);
+
+  useEffect(async () => {
+    if (state.memberList) {
+      await updateProject();
+    }
+  }, [state.memberList]);
+
+
+
+  const addNewMember = (member) => {
+    setState({ memberList: [...state.memberList, member] });
+  };
+
+  const onDeleteItem = (item) => {
+    setState({
+      memberList: state.memberList.filter((member) => member !== item),
+    });
+  };
+  const onMemberClick = (item) => {
+    //Enable editing the member
+    setState({
+      editMode: true,
+      isEditing: true,
+      editingIndex: state.memberList.findIndex((i) => i === item),
+    });
+  };
+
+  const updateProject = async () => {
+    //Update the db
+    try {
+      let result = await db.updateProject({
+        projectName: project.projectName,
+        member: state.memberList,
+      });
+
+      //Check if the update worked
+      if (!result.success) throw { message: result.message };
+
+      await refreshProjects(); //Refresh the projects stored as they were modified
+      displayPopup("Successfully saved member");
+    } catch (err) {
+      displayPopup(`Error saving member ${err.message}`);
+
+      console.log(`Project update failed: ${err}`);
+    }
+  };
+
+  const renderList = () => {
+    if (state.memberList) {
+      return state.memberList.map((item) => (
+        <ListItem button onClick={() => onMemberClick(item)}>
+          <ListItemText
+            primary={item.memberEmail}
+            style={{ maxWidth: "10px", marginRight: "50px" }}
+          />
+          <ListItemText primary={item.memberEmail} style={{ flex: 12 }} />
+          <ListItemSecondaryAction>
+            <IconButton
+              edge="end"
+              aria-label="delete"
+              onClick={() => onDeleteItem(item)}
+            >
+              <DeleteIcon />
+            </IconButton>
+          </ListItemSecondaryAction>
+        </ListItem>
+      ));
+    }
+  };
+
+  const isInvalidMember = () => {
+    return (
+      newMember.email === "" ||
+      newMember.email === undefined
+    );
+  };
+
+  const onAddOrUpdateMember = async () => {
+    console.log(isInvalidMember);
+    if (!isInvalidMember) {
+      setState({
+        addMode: false,
+        editMode: false,
+        addMemberError: "",
+        isEditing: false,
+        editingIndex: null,
+      });
+      if (state.isEditing) {
+        let memberList = state.memberList;
+        memberList[state.editingIndex] = newMember;
+
+        setState({
+          isEditing: false,
+          editingIndex: null,
+        });
+      }else await addNewMember(newMember);
+      setNewMember({
+        email: "",
+        firstName: "",
+        lastName: "",
+      });
+    } else {
+      setState({
```

Comment:
I have no idea why this is always fiering

Reply:
That was one of the bugs that I think I found in my code and updated in the last pr